### DISCUSSION TITLE
Autoscaling of runners doesn't seem to work properly

### DISCUSSION THREAD

**Discussion by rahul9170** [actions/actions-runner-controller#1408] [Questions]
Hello,

I am trying to setup the controller v0.23.0 and facing issue with auto scaling have below args passed to controller deployment.yaml

<img width="279" alt="image" src="https://user-images.githubusercontent.com/61755746/166242493-6d253234-25de-4e8c-ad25-010270cb3038.png">

RunnderDeployment.yaml (Replicas=0 has been set)

<img width="419" alt="image" src="https://user-images.githubusercontent.com/61755746/166242809-b8b63200-cfeb-4cdc-8d54-f6359081b5f3.png">

HorizontalRunnerAutoscaler.yaml (status.desiredReplicas=0)

<img width="434" alt="image" src="https://user-images.githubusercontent.com/61755746/166242926-03b7085b-1194-4efd-9ec7-93e6ebaa4e15.png">

After a successful automatic scale up of 1 runner, it should scale down as per **scaleDownDelaySecondsAfterScaleOut** setting but the runner pod continues to run. The controller logs also suggest the same but pod continues to run

<img width="1785" alt="image" src="https://user-images.githubusercontent.com/61755746/166243510-f879bec7-4a7a-436a-ae1b-dfb5ad769b98.png">

Can you please help with this issue?

**Comment by rahul9170** [comment]
Also I get below entry in the logs

<img width="1791" alt="image" src="https://user-images.githubusercontent.com/61755746/166244388-1640584c-c603-461e-8135-8a12ad7f842f.png">

**Comment by toast-gear** [comment]
You've not really provided enough inforamation for anyone to help you, you don't need to hide log lines. The controller logs reasons why it isn't scaling something down. What additional log information is there that relates to scaling down decisions?

**Comment by mumoshu** [comment]
>  it should scale down as per scaleDownDelaySecondsAfterScaleOut

This perception can be problematic. You have `--sync-period=1m` so ARC should only able to sync the desired replicas according to the content returned by GitHub API approx once per 1m.

> controller deployment.yaml

Also, how did you deploy ARC?
Did you really deployed ARC using kubectl or kustomize without helm even though you seem to be using helm for deploying runners?
